### PR TITLE
feat: full pdf summary of applications (HL-708, HL-903)

### PR DESCRIPTION
**Pull request by sirtawast**
## Description :sparkles:

Tests fail for some reason on the other PR. Try here instead.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2524.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2524.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2524.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2524.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2524.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2524.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2524.dev.hel.ninja :laughing::tada::tada::tada:

File: backend/benefit/helsinkibenefit/urls.py [hunk 1]
```diff
@@ -88,6 +88,7 @@
     path(
         "v1/handlerapplications/<str:application_id>/review/", ReviewStateView.as_view()
     ),
+    path("v1/print/<str:pk>/", application_views.PrintDetail.as_view()),
```
Comment (rikuke) [inline backend/benefit/helsinkibenefit/urls.py:91]: If you put name="foo" here you can get the route by name later in tests for example like
reverse("foo", kwargs={"yourparam": "bar"}).

File: backend/benefit/applications/tests/test_applications_api.py [hunk 2]
```diff
@@ -2219,6 +2219,40 @@ def test_handler_application_filter_archived(handler_api_client):
...
+
+def test_application_pdf_print_denied(api_client, anonymous_client):
+    settings.NEXT_PUBLIC_MOCK_FLAG = False  # noqa
+    reseed(12345)
+
+    application = DecidedApplicationFactory()
+
+    # Cannot access anonymously
+    response = anonymous_client.get(
+        f"/v1/print/{application.id}/",
```
Comment (rikuke) [inline backend/benefit/applications/tests/test_applications_api.py:2242]: Here for example the named route could be used.
`reverse("print_application", kwargs={"id":  application.id}).`

File: backend/benefit/applications/services/app_generated_files.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        "de_minimis_amount": total_de_minimis_amount(application),
+        "de_minimis_aid_set": application.de_minimis_aid_set.all(),
+        "versions": {
+            "application": count_version_number(len(application.history.all()))
+        },
+    }
+
+
+def generate_application_summary_file(application, request=None) -> bytes:
+    def generate_summary_pdf(context) -> bytes:
```
Comment (rikuke) [inline backend/benefit/applications/services/app_generated_files.py:35]: Can something co wrong here so badly that it crashes the app? Meaning is there an exception that can be catched and handled?

**Review by rikuke — APPROVED** [review]

**Pull request merged**